drivers/ethernet/Kconfig.sam_gmac

```diff
@@ -77,6 +77,14 @@ config ETH_SAM_GMAC_IRQ_PRI
 	help
 	  IRQ priority of Ethernet device
 
+config ETH_SAM_GMAC_MONITOR_PERIOD
```

Comment:
I understand that this is required but we should also add support for phy link interrupts wherever possible. Changing of PHY Link is not something that happens all the time and so a work (per instance) is just an overhead on the system. 

Where link interrupts are not available, we should fallback to the polled approach..

Reply:
While I agree that using PHY interrupt is better, as I already mentioned in https://github.com/zephyrproject-rtos/zephyr/issues/23386, that is not part of the standard PHY-MAC interface and overhead of periodic polling is minimal, so I decided to not bother with it for now (in fact, most Ethernet drivers in Zephyr poll, instead of using the PHY interrupt).

If you want to implement link status change detection using PHY interrupt, feel free to do so.